src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -921,17 +929,47 @@ private CloudServiceInstance getServiceInstanceIfRequired(String serviceInstance
         return serviceInstance;
     }
 
+    @Override
+    public CloudServiceBinding getServiceBinding(UUID serviceBindingGuid) {
+        // TODO: Refactor and use serviceBindingsV3.get() when https://github.com/cloudfoundry/cf-java-client/pull/1171 is merged and
+        // released
+        return fetch(() -> getServiceBindingResource(serviceBindingGuid), ImmutableRawCloudServiceBinding::of);
+    }
+
+    private Mono<ServiceBindingResource> getServiceBindingResource(UUID serviceBindingGuid) {
+        IntFunction<ListServiceBindingsRequest> pageRequestSupplier = page -> ListServiceBindingsRequest.builder()
+                                                                                                        .page(page)
+                                                                                                        .id(serviceBindingGuid.toString())
+                                                                                                        .build();
+        return PaginationUtils.requestClientV3Resources(page -> delegate.serviceBindingsV3()
+                                                                        .list(pageRequestSupplier.apply(page)))
+                              .singleOrEmpty();
+    }
+
     @Override
     public List<CloudServiceBinding> getServiceAppBindings(UUID serviceInstanceGuid) {
         return fetchList(() -> getServiceBindingResourcesByServiceInstanceGuid(serviceInstanceGuid), ImmutableRawCloudServiceBinding::of);
     }
 
+    @Override
+    public List<CloudServiceBinding> getAppBindings(String applicationName) {
```

Comment:
Can we keep same approach as `getServiceAppBindings` and `getServiceBindingForApplication` and pass UUID only. In this way we'll have more control over made requests at levels above.